File: ComponentCompiler/src/main/java/com/xiaojinzi/component/bean/InterceptorBean.java
package com.xiaojinzi.component.bean;

import com.xiaojinzi.component.anno.support.CheckClassName;

import javax.lang.model.element.Element;

/**
 * time   : 2018/12/26
 *
 * @author : xiaojinzi
 */
@CheckClassName
public class InterceptorBean {

    public static final int GLOBAL_INTERCEPTOR = 0;
    public static final int NORMAL_INTERCEPTOR = 1;

    /**
     * 0 全局拦截器
     * 1 普通拦截器
     */
    public final int type;

    public final Element element;

    public final int priority;

    public final String name;

    public InterceptorBean(int type, Element element, int priority, String name) {
        this.type = type;
        this.element = element;
        this.priority = priority;
        this.name = name;
    }

}

File: ComponentImpl/src/main/java/com/xiaojinzi/component/impl/interceptor/InterceptorBean.kt
package com.xiaojinzi.component.impl.interceptor

import com.xiaojinzi.component.impl.RouterInterceptor

/**
 * time   : 2018/12/26
 *
 * @author : xiaojinzi
 */
data class InterceptorBean(
        /**
         * 拦截器
         */
        val interceptor: RouterInterceptor,
        /**
         * 优先级
         */
        val priority: Int
)